src/ngraph/runtime/gpu/gpu_cuda_kernel_builder.cpp

```diff
@@ -33,14 +33,13 @@ void runtime::gpu::CudaKernelBuilder::get_elementwise_op(codegen::CodeWriter& wr
         writer << data_types[i] << "* in" << i << ", ";
     }
     writer << data_types[num_inputs] << "* out, "
-           << "size_t n)\n";
-    writer << "{\n";
-    writer.indent++;
+           << "uint32_t n)\n";
+    writer.block_begin();
     {
-        writer << "size_t tid = blockIdx.x * blockDim.x + threadIdx.x; \n";
-        writer << "if (tid < n)\n";
-        writer << "{\n";
-        writer.indent++;
+        writer << "uint32_t tid = blockIdx.x * blockDim.x + threadIdx.x; \n";
+        writer << "uint32_t step = gridDim.x * blockDim.x; \n";
+        writer << "for ( ;tid < n; tid += step)\n";
```

Comment:
Have you experimented with unrolling this loop? It would require an extra parameter to come into the cuda_kernel_builder function.

Reply:
Here it's hard to unroll, since the initial value of `tid` is not fixed. So thread will have different number of loop. Plus `step` can also change.

Reply:
Not too hard, the information is known from when we calculate the number of blocks in the grid. So the emitter could take an extra parameter that would indicate the number of blocks, and from that you can calculate the number of steps needed.

It would have the side effect of making the kernel semi-shape dependent, but it could improve the performance. It's not critical for this PR, but I am interested in the impact it might have.

Reply:
Ah, but you maybe need to either pad or add a branch. Anyway, approved, just wondered if you had experimented with it at all.